stdlib/influxdata/influxdb/source.go

```diff
@@ -75,7 +76,12 @@ func (s *source) run(ctx context.Context) error {
 		return err
 	}
 
-	client, err := s.deps.HTTPClient()
+	var client fluxhttp.Client
+	if *s.spec.GetHost() == "" {
+		client, err = s.deps.PrivateHTTPClient()
+	} else {
+		client, err = s.deps.HTTPClient()
+	}
```

Comment:
I don't see where a default `host` would be provided here, so I think that this is dead code; `host` will always be defined. I believe that `validateHost` would produce an error otherwise.

This implementation of `from` is for when we it's invoked with an explicit host, so by default, the user is supposed to have provided one. There is planner rule in cloud that will convert a `from` that is destined for a local cluster.

Reply:
If it's dead code, I'll just remove it. Maybe the converted `from()` has a section of code where this branch would be more appropriate?